script_1.js

```diff
@@ -0,0 +1,185 @@
+console.log("hello world");
+
+document.addEventListener("DOMContentLoaded", init);
+
+function init() {
+  let data;
+
+  fetch(
+    "https://raw.githubusercontent.com/catalina-tech/catalina-tech.github.oi/main/data/apiData.json"
+  )
+    .then((response) => {
+      return response.json();
+    })
+    .then((jsonData) => {
+      data = jsonData;
+      updatedDOM(data);
+      const questions = data;
+      startQuiz(questions);
+    })
+    .catch((error) => console.log("there is something wrong", error))
+    .finally(() => (loading = false));
+}
+
+function updatedDOM(questions) {
+  const playerNameElement = document.getElementById("playerName");
+  const quizSection = document.getElementById("quizSection");
+  const playerNameInput = document.getElementById("playerNameInput");
+  const startButton = document.getElementById("startButton");
+  const questionElement = document.getElementById("question");
+  const answerButtons = document.getElementById("answerButtons");
+  const nextButton = document.getElementById("btn-next");
+  const shuffleButton = document.getElementById("shuffleButton");
+  const explanationElement = document.getElementById("explanation");
+  const divApp = document.getElementById("app");
+
+  playerNameElement.textContent = playerNameInput;
+
+  let currentQuestionIndex = 0;
+  let score = [0];
+  let player = "";
+  let currentPlayerIndex = 0;
+
+  startButton.addEventListener("click", startButtonClick);
+
+  function startButtonClick() {
+    const playerName = playerNameInput.value.trim();
+
+    if (playerName) {
+      localStorage.setItem("playerName", playerName);
+      player = playerName;
+      divApp.classList.add("hidden");
+      quizSection.classList.remove("hidden");
+      playerNameElement.textContent = "PLAYER: " + playerName;
+      playerNameElement.style.textTransform = "uppercase";
+      startQuiz(questions);
+    } else {
+      alert("Please enter your name.");
+    }
+  }
+
+  function startQuiz(questions) {
+    currentQuestionIndex = 0;
+    currentPlayerIndex = 0;
+    score = [0];
+    nextButton.innerHTML = "Next";
+    shuffleButton.innerHTML = "Shuffle Answers";
+    shuffleButton.style.display = "block";
+    showQuestion(questions);
+  }
+
+  function showQuestion(questions) {
```

Comment:
When you have a long function like this, it's a good idea to either organise it into blocks of code, separated by an empty line, and with a comment before each block telling briefly what the block does. Or, you could organise it into more functions.

The goal here is to make the function easy to read.